lib/spack/spack/test/data/config/config.yaml

```diff
@@ -14,4 +14,5 @@ config:
   checksum: true
   dirty: false
   module_roots:
-    tcl:    $spack/share/spack/modules
+    tcl: {0}
+    lmod: {1}
```

Comment:
Oh finally!  I guess this has been an issue for quite some time; we just never bothered to fix it in the tests.